client/authority-discovery/src/worker.rs

```diff
@@ -133,7 +132,7 @@ where
 	//   - Some(vec![a, b, c, ...]): Valid addresses were specified.
 	sentry_nodes: Option<Vec<Multiaddr>>,
 	/// Channel we receive Dht events on.
-	dht_event_rx: Pin<Box<dyn Stream<Item = DhtEvent> + Send>>,
+	dht_event_rx: Pin<Box<dyn Stream<Item = DhtEvent> + Send + Sync>>,
```

Comment:
Why add the trait bound `Sync` to `dht_event_rx`? When do you need to share references to `dht_event_rx` across threads?

Reply:
I removed the trait bound. However, when compiling `bin/node/cli` i get the following error:
```
error[E0277]: `(dyn futures::Stream<Item = sc_network::DhtEvent> + std::marker::Send + 'static)` cannot be shared between threads safely
   --> bin/node/cli/src/service.rs:278:31
    |
278 |         task_manager.spawn_handle().spawn("authority-discovery-worker", authority_discovery_worker.run());
    |                                     ^^^^^ `(dyn futures::Stream<Item = sc_network::DhtEvent> + std::marker::Send + 'static)` cannot be shared between threads safely
    |
    = help: the trait `std::marker::Sync` is not implemented for `(dyn futures::Stream<Item = sc_network::DhtEvent> + std::marker::Send + 'static)`
    = note: required because of the requirements on the impl of `std::marker::Sync` for `std::ptr::Unique<(dyn futures::Stream<Item = sc_network::DhtEvent> + std::marker::Send + 'static)>`
    = note: required because it appears within the type `std::boxed::Box<(dyn futures::Stream<Item = sc_network::DhtEvent> + std::marker::Send + 'static)>`
    = note: required because it appears within the type `std::pin::Pin<std::boxed::Box<(dyn futures::Stream<Item = sc_network::DhtEvent> + std::marker::Send + 'static)>>`
    = note: required because it appears within the type `sc_authority_discovery::Worker<sc_service::client::Client<sc_client_db::Backend<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>, sc_service::client::LocalCallExecutor<sc_client_db::Backend<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>, node_executor::NativeExecutor<node_executor::Executor>>, sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>, node_runtime::RuntimeApi>, sc_network::NetworkService<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>, sp_core::H256>, sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>`
    = note: required because of the requirements on the impl of `std::marker::Send` for `&sc_authority_discovery::Worker<sc_service::client::Client<sc_client_db::Backend<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>, sc_service::client::LocalCallExecutor<sc_client_db::Backend<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>, node_executor::NativeExecutor<node_executor::Executor>>, sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>, node_runtime::RuntimeApi>, sc_network::NetworkService<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>, sp_core::H256>, sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>`
    = note: required because it appears within the type `for<'r, 's, 't0, 't1, 't2, 't3, 't4, 't5, 't6, 't7, 't8, 't9, 't10, 't11, 't12, 't13, 't14, 't15, 't16, 't17, 't18, 't19, 't20, 't21, 't22, 't23, 't24, 't25, 't26, 't27, 't28, 't29, 't30, 't31, 't32, 't33, 't34, 't35, 't36> {std::future::ResumeTy, sc_authority_discovery::Worker<sc_service::client::Client<sc_client_db::Backend<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>, sc_service::client::LocalCallExecutor<sc_client_db::Backend<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>, node_executor::NativeExecutor<node_executor::Executor>>, sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>, node_runtime::RuntimeApi>, sc_network::NetworkService<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>, sp_core::H256>, sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>, futures::future::Fuse<futures::stream::Next<'r, std::pin::Pin<std::boxed::Box<(dyn futures::Stream<Item = sc_network::DhtEvent> + std::marker::Send + 's)>>>>, futures::future::Fuse<futures::stream::Next<'t0, futures::stream::Fuse<futures::channel::mpsc::Receiver<sc_authority_discovery::ServicetoWorkerMsg>>>>, futures::future::Fuse<futures::stream::Next<'t1, std::boxed::Box<(dyn futures::Stream<Item = ()> + std::marker::Send + std::marker::Sync + std::marker::Unpin + 't2)>>>, [closure@sc_authority_discovery::Worker::<Client, Network, Block>::run::{{closure}}#0::{{closure}}#0 0:&'t3 mut futures::future::Fuse<futures::stream::Next<'t4, std::pin::Pin<std::boxed::Box<(dyn futures::Stream<Item = sc_network::DhtEvent> + std::marker::Send + 't5)>>>>, 1:&'t6 mut futures::future::Fuse<futures::stream::Next<'t7, futures::stream::Fuse<futures::channel::mpsc::Receiver<sc_authority_discovery::ServicetoWorkerMsg>>>>, 2:&'t8 mut futures::future::Fuse<futures::stream::Next<'t9, std::boxed::Box<(dyn futures::Stream<Item = ()> + std::marker::Send + std::marker::Sync + std::marker::Unpin + 't10)>>>, 3:&'t11 mut futures::future::Fuse<futures::stream::Next<'t12, std::boxed::Box<(dyn futures::Stream<Item = ()> + std::marker::Send + std::marker::Sync + std::marker::Unpin + 't13)>>>, 4:&'t14 mut futures::future::Fuse<futures::stream::Next<'t15, std::boxed::Box<(dyn futures::Stream<Item = ()> + std::marker::Send + std::marker::Sync + std::marker::Unpin + 't16)>>>], futures::future::PollFn<[closure@sc_authority_discovery::Worker::<Client, Network, Block>::run::{{closure}}#0::{{closure}}#0 0:&'t17 mut futures::future::Fuse<futures::stream::Next<'t18, std::pin::Pin<std::boxed::Box<(dyn futures::Stream<Item = sc_network::DhtEvent> + std::marker::Send + 't19)>>>>, 1:&'t20 mut futures::future::Fuse<futures::stream::Next<'t21, futures::stream::Fuse<futures::channel::mpsc::Receiver<sc_authority_discovery::ServicetoWorkerMsg>>>>, 2:&'t22 mut futures::future::Fuse<futures::stream::Next<'t23, std::boxed::Box<(dyn futures::Stream<Item = ()> + std::marker::Send + std::marker::Sync + std::marker::Unpin + 't24)>>>, 3:&'t25 mut futures::future::Fuse<futures::stream::Next<'t26, std::boxed::Box<(dyn futures::Stream<Item = ()> + std::marker::Send + std::marker::Sync + std::marker::Unpin + 't27)>>>, 4:&'t28 mut futures::future::Fuse<futures::stream::Next<'t29, std::boxed::Box<(dyn futures::Stream<Item = ()> + std::marker::Send + std::marker::Sync + std::marker::Unpin + 't30)>>>]>, (), sc_authority_discovery::Worker<Client, Network, Block>::run::{{closure}}#0::__PrivResult<std::option::Option<sc_network::DhtEvent>, std::option::Option<sc_authority_discovery::ServicetoWorkerMsg>, std::option::Option<()>, std::option::Option<()>, std::option::Option<()>>, std::option::Option<sc_network::DhtEvent>, sc_network::DhtEvent, &'t31 mut sc_authority_discovery::Worker<sc_service::client::Client<sc_client_db::Backend<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>, sc_service::client::LocalCallExecutor<sc_client_db::Backend<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>, node_executor::NativeExecutor<node_executor::Executor>>, sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>, node_runtime::RuntimeApi>, sc_network::NetworkService<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>, sp_core::H256>, sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>, impl futures::Future, std::option::Option<sc_authority_discovery::ServicetoWorkerMsg>, sc_authority_discovery::ServicetoWorkerMsg, &'t33 sc_authority_discovery::Worker<sc_service::client::Client<sc_client_db::Backend<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>, sc_service::client::LocalCallExecutor<sc_client_db::Backend<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>, node_executor::NativeExecutor<node_executor::Executor>>, sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>, node_runtime::RuntimeApi>, sc_network::NetworkService<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>, sp_core::H256>, sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>, impl futures::Future, impl futures::Future, impl futures::Future}`
    = note: required because it appears within the type `[static generator@sc_authority_discovery::Worker::<Client, Network, Block>::run::{{closure}}#0 0:sc_authority_discovery::Worker<sc_service::client::Client<sc_client_db::Backend<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>, sc_service::client::LocalCallExecutor<sc_client_db::Backend<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>, node_executor::NativeExecutor<node_executor::Executor>>, sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>, node_runtime::RuntimeApi>, sc_network::NetworkService<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>, sp_core::H256>, sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>> for<'r, 's, 't0, 't1, 't2, 't3, 't4, 't5, 't6, 't7, 't8, 't9, 't10, 't11, 't12, 't13, 't14, 't15, 't16, 't17, 't18, 't19, 't20, 't21, 't22, 't23, 't24, 't25, 't26, 't27, 't28, 't29, 't30, 't31, 't32, 't33, 't34, 't35, 't36> {std::future::ResumeTy, sc_authority_discovery::Worker<sc_service::client::Client<sc_client_db::Backend<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>, sc_service::client::LocalCallExecutor<sc_client_db::Backend<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>, node_executor::NativeExecutor<node_executor::Executor>>, sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>, node_runtime::RuntimeApi>, sc_network::NetworkService<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>, sp_core::H256>, sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>, futures::future::Fuse<futures::stream::Next<'r, std::pin::Pin<std::boxed::Box<(dyn futures::Stream<Item = sc_network::DhtEvent> + std::marker::Send + 's)>>>>, futures::future::Fuse<futures::stream::Next<'t0, futures::stream::Fuse<futures::channel::mpsc::Receiver<sc_authority_discovery::ServicetoWorkerMsg>>>>, futures::future::Fuse<futures::stream::Next<'t1, std::boxed::Box<(dyn futures::Stream<Item = ()> + std::marker::Send + std::marker::Sync + std::marker::Unpin + 't2)>>>, [closure@sc_authority_discovery::Worker::<Client, Network, Block>::run::{{closure}}#0::{{closure}}#0 0:&'t3 mut futures::future::Fuse<futures::stream::Next<'t4, std::pin::Pin<std::boxed::Box<(dyn futures::Stream<Item = sc_network::DhtEvent> + std::marker::Send + 't5)>>>>, 1:&'t6 mut futures::future::Fuse<futures::stream::Next<'t7, futures::stream::Fuse<futures::channel::mpsc::Receiver<sc_authority_discovery::ServicetoWorkerMsg>>>>, 2:&'t8 mut futures::future::Fuse<futures::stream::Next<'t9, std::boxed::Box<(dyn futures::Stream<Item = ()> + std::marker::Send + std::marker::Sync + std::marker::Unpin + 't10)>>>, 3:&'t11 mut futures::future::Fuse<futures::stream::Next<'t12, std::boxed::Box<(dyn futures::Stream<Item = ()> + std::marker::Send + std::marker::Sync + std::marker::Unpin + 't13)>>>, 4:&'t14 mut futures::future::Fuse<futures::stream::Next<'t15, std::boxed::Box<(dyn futures::Stream<Item = ()> + std::marker::Send + std::marker::Sync + std::marker::Unpin + 't16)>>>], futures::future::PollFn<[closure@sc_authority_discovery::Worker::<Client, Network, Block>::run::{{closure}}#0::{{closure}}#0 0:&'t17 mut futures::future::Fuse<futures::stream::Next<'t18, std::pin::Pin<std::boxed::Box<(dyn futures::Stream<Item = sc_network::DhtEvent> + std::marker::Send + 't19)>>>>, 1:&'t20 mut futures::future::Fuse<futures::stream::Next<'t21, futures::stream::Fuse<futures::channel::mpsc::Receiver<sc_authority_discovery::ServicetoWorkerMsg>>>>, 2:&'t22 mut futures::future::Fuse<futures::stream::Next<'t23, std::boxed::Box<(dyn futures::Stream<Item = ()> + std::marker::Send + std::marker::Sync + std::marker::Unpin + 't24)>>>, 3:&'t25 mut futures::future::Fuse<futures::stream::Next<'t26, std::boxed::Box<(dyn futures::Stream<Item = ()> + std::marker::Send + std::marker::Sync + std::marker::Unpin + 't27)>>>, 4:&'t28 mut futures::future::Fuse<futures::stream::Next<'t29, std::boxed::Box<(dyn futures::Stream<Item = ()> + std::marker::Send + std::marker::Sync + std::marker::Unpin + 't30)>>>]>, (), sc_authority_discovery::Worker<Client, Network, Block>::run::{{closure}}#0::__PrivResult<std::option::Option<sc_network::DhtEvent>, std::option::Option<sc_authority_discovery::ServicetoWorkerMsg>, std::option::Option<()>, std::option::Option<()>, std::option::Option<()>>, std::option::Option<sc_network::DhtEvent>, sc_network::DhtEvent, &'t31 mut sc_authority_discovery::Worker<sc_service::client::Client<sc_client_db::Backend<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>, sc_service::client::LocalCallExecutor<sc_client_db::Backend<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>, node_executor::NativeExecutor<node_executor::Executor>>, sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>, node_runtime::RuntimeApi>, sc_network::NetworkService<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>, sp_core::H256>, sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>, impl futures::Future, std::option::Option<sc_authority_discovery::ServicetoWorkerMsg>, sc_authority_discovery::ServicetoWorkerMsg, &'t33 sc_authority_discovery::Worker<sc_service::client::Client<sc_client_db::Backend<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>, sc_service::client::LocalCallExecutor<sc_client_db::Backend<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>, node_executor::NativeExecutor<node_executor::Executor>>, sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>, node_runtime::RuntimeApi>, sc_network::NetworkService<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>, sp_core::H256>, sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>, impl futures::Future, impl futures::Future, impl futures::Future}]`
    = note: required because it appears within the type `std::future::from_generator::GenFuture<[static generator@sc_authority_discovery::Worker::<Client, Network, Block>::run::{{closure}}#0 0:sc_authority_discovery::Worker<sc_service::client::Client<sc_client_db::Backend<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>, sc_service::client::LocalCallExecutor<sc_client_db::Backend<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>, node_executor::NativeExecutor<node_executor::Executor>>, sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>, node_runtime::RuntimeApi>, sc_network::NetworkService<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>, sp_core::H256>, sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>> for<'r, 's, 't0, 't1, 't2, 't3, 't4, 't5, 't6, 't7, 't8, 't9, 't10, 't11, 't12, 't13, 't14, 't15, 't16, 't17, 't18, 't19, 't20, 't21, 't22, 't23, 't24, 't25, 't26, 't27, 't28, 't29, 't30, 't31, 't32, 't33, 't34, 't35, 't36> {std::future::ResumeTy, sc_authority_discovery::Worker<sc_service::client::Client<sc_client_db::Backend<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>, sc_service::client::LocalCallExecutor<sc_client_db::Backend<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>, node_executor::NativeExecutor<node_executor::Executor>>, sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>, node_runtime::RuntimeApi>, sc_network::NetworkService<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>, sp_core::H256>, sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>, futures::future::Fuse<futures::stream::Next<'r, std::pin::Pin<std::boxed::Box<(dyn futures::Stream<Item = sc_network::DhtEvent> + std::marker::Send + 's)>>>>, futures::future::Fuse<futures::stream::Next<'t0, futures::stream::Fuse<futures::channel::mpsc::Receiver<sc_authority_discovery::ServicetoWorkerMsg>>>>, futures::future::Fuse<futures::stream::Next<'t1, std::boxed::Box<(dyn futures::Stream<Item = ()> + std::marker::Send + std::marker::Sync + std::marker::Unpin + 't2)>>>, [closure@sc_authority_discovery::Worker::<Client, Network, Block>::run::{{closure}}#0::{{closure}}#0 0:&'t3 mut futures::future::Fuse<futures::stream::Next<'t4, std::pin::Pin<std::boxed::Box<(dyn futures::Stream<Item = sc_network::DhtEvent> + std::marker::Send + 't5)>>>>, 1:&'t6 mut futures::future::Fuse<futures::stream::Next<'t7, futures::stream::Fuse<futures::channel::mpsc::Receiver<sc_authority_discovery::ServicetoWorkerMsg>>>>, 2:&'t8 mut futures::future::Fuse<futures::stream::Next<'t9, std::boxed::Box<(dyn futures::Stream<Item = ()> + std::marker::Send + std::marker::Sync + std::marker::Unpin + 't10)>>>, 3:&'t11 mut futures::future::Fuse<futures::stream::Next<'t12, std::boxed::Box<(dyn futures::Stream<Item = ()> + std::marker::Send + std::marker::Sync + std::marker::Unpin + 't13)>>>, 4:&'t14 mut futures::future::Fuse<futures::stream::Next<'t15, std::boxed::Box<(dyn futures::Stream<Item = ()> + std::marker::Send + std::marker::Sync + std::marker::Unpin + 't16)>>>], futures::future::PollFn<[closure@sc_authority_discovery::Worker::<Client, Network, Block>::run::{{closure}}#0::{{closure}}#0 0:&'t17 mut futures::future::Fuse<futures::stream::Next<'t18, std::pin::Pin<std::boxed::Box<(dyn futures::Stream<Item = sc_network::DhtEvent> + std::marker::Send + 't19)>>>>, 1:&'t20 mut futures::future::Fuse<futures::stream::Next<'t21, futures::stream::Fuse<futures::channel::mpsc::Receiver<sc_authority_discovery::ServicetoWorkerMsg>>>>, 2:&'t22 mut futures::future::Fuse<futures::stream::Next<'t23, std::boxed::Box<(dyn futures::Stream<Item = ()> + std::marker::Send + std::marker::Sync + std::marker::Unpin + 't24)>>>, 3:&'t25 mut futures::future::Fuse<futures::stream::Next<'t26, std::boxed::Box<(dyn futures::Stream<Item = ()> + std::marker::Send + std::marker::Sync + std::marker::Unpin + 't27)>>>, 4:&'t28 mut futures::future::Fuse<futures::stream::Next<'t29, std::boxed::Box<(dyn futures::Stream<Item = ()> + std::marker::Send + std::marker::Sync + std::marker::Unpin + 't30)>>>]>, (), sc_authority_discovery::Worker<Client, Network, Block>::run::{{closure}}#0::__PrivResult<std::option::Option<sc_network::DhtEvent>, std::option::Option<sc_authority_discovery::ServicetoWorkerMsg>, std::option::Option<()>, std::option::Option<()>, std::option::Option<()>>, std::option::Option<sc_network::DhtEvent>, sc_network::DhtEvent, &'t31 mut sc_authority_discovery::Worker<sc_service::client::Client<sc_client_db::Backend<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>, sc_service::client::LocalCallExecutor<sc_client_db::Backend<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>, node_executor::NativeExecutor<node_executor::Executor>>, sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>, node_runtime::RuntimeApi>, sc_network::NetworkService<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>, sp_core::H256>, sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>, impl futures::Future, std::option::Option<sc_authority_discovery::ServicetoWorkerMsg>, sc_authority_discovery::ServicetoWorkerMsg, &'t33 sc_authority_discovery::Worker<sc_service::client::Client<sc_client_db::Backend<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>, sc_service::client::LocalCallExecutor<sc_client_db::Backend<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>, node_executor::NativeExecutor<node_executor::Executor>>, sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>, node_runtime::RuntimeApi>, sc_network::NetworkService<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>, sp_core::H256>, sp_runtime::generic::Block<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>, impl futures::Future, impl futures::Future, impl futures::Future}]>`
    = note: required because it appears within the type `impl futures::Future`
    = note: required because it appears within the type `impl futures::Future`
```

I am gonna have to investigate what is causing this.